lib/trino-parquet/src/main/java/io/trino/parquet/reader/ParquetReader.java

```diff
@@ -377,6 +381,11 @@ private ColumnChunk readPrimitive(PrimitiveField field)
         return columnChunk;
     }
 
+    public Set<String> getCodecsUsed()
+    {
+        return codecsUsed;
```

Comment:
this returns mutable internal state.

if https://github.com/trinodb/trino/pull/12551/files#r882352097 possible, this will make this immutable
otherwise, make a copy